specs/phase0/fork-choice.md

```diff
@@ -218,17 +261,29 @@ def filter_block_tree(store: Store, block_root: Root, blocks: Dict[Root, BeaconB
             return True
         return False
 
-    # If leaf block, check finalized/justified checkpoints as matching latest.
-    head_state = store.block_states[block_root]
+    current_epoch = compute_epoch_at_slot(get_current_slot(store))
+    voting_source = get_voting_source(store, block_root)
 
+    # The voting source should be at the same height as the store's justified checkpoint
     correct_justified = (
         store.justified_checkpoint.epoch == GENESIS_EPOCH
-        or head_state.current_justified_checkpoint == store.justified_checkpoint
+        or voting_source.epoch == store.justified_checkpoint.epoch
     )
+
+    # If the previous epoch is justified, the block should be pulled-up. In this case, check that unrealized
+    # justification is higher than the store and that the voting source is not more than two epochs ago
```

Comment:
Can you help explain this a bit more? I'm curious about the intuition behind it.

It seems to me that this contradict to the HLMD-GHOST in https://arxiv.org/pdf/2003.03052.pdf
```
L′ ← set of leaf blocks Bl in G such that (Bj , j) ∈ J(ffgview(Bl))
```